npm/pkg/dataplane/ipsets/ipset.go

```diff
@@ -0,0 +1,191 @@
+	NetPolReference   map[string]struct{}
+	IpsetReferCount   int
+}
+
+type SetType int32
+
+const (
+	Unknown                  SetType = 0
+	NameSpace                SetType = 1
+	KeyLabelOfNameSpace      SetType = 2
+	KeyValueLabelOfNameSpace SetType = 3
+	KeyLabelOfPod            SetType = 4
+	KeyValueLabelOfPod       SetType = 5
+	NamedPorts               SetType = 6
+	NestedLabelOfPod         SetType = 7
+	CIDRBlocks               SetType = 8
+)
+
+var SetTypeName = map[int32]string{
+	0: "Unknown",
+	1: "NameSpace",
+	2: "KeyLabelOfNameSpace",
+	3: "KeyValueLabelOfNameSpace",
+	4: "KeyLabelOfPod",
+	5: "KeyValueLabelOfPod",
+	6: "NamedPorts",
+	7: "NestedLabelOfPod",
+	8: "CIDRBlocks",
+}
+
+var SetTypeValue = map[string]int32{
+	"Unknown":                  0,
+	"NameSpace":                1,
+	"KeyLabelOfNameSpace":      2,
+	"KeyValueLabelOfNameSpace": 3,
+	"KeyLabelOfPod":            4,
+	"KeyValueLabelOfPod":       5,
+	"NamedPorts":               6,
+	"NestedLabelOfPod":         7,
+	"CIDRBlocks":               8,
+}
+
+func (x SetType) String() string {
+	return SetTypeName[int32(x)]
+}
+
+func GetSetType(x string) SetType {
+	return SetType(SetTypeValue[x])
+}
+
+type SetKind string
+
+const (
+	ListSet SetKind = "list"
+	HashSet SetKind = "set"
+)
+
+func NewIPSet(name string, setType SetType) *IPSet {
+	set := &IPSet{
+		Name:              name,
+		HashedName:        util.GetHashedName(name),
+		Type:              setType,
+		SelectorReference: make(map[string]struct{}),
+		NetPolReference:   make(map[string]struct{}),
+		IpsetReferCount:   0,
+	}
+	if getSetKind(set) == HashSet {
+		set.IPPodKey = make(map[string]string)
+	} else {
+		set.MemberIPSets = make(map[string]*IPSet)
+	}
+	return set
+}
+
+func (set *IPSet) GetSetContents() ([]string, error) {
+	contents := make([]string, 0)
+	setType := getSetKind(set)
+	switch setType {
+	case HashSet:
+		for podIP := range set.IPPodKey {
+			contents = append(contents, podIP)
+		}
+		return contents, nil
+	case ListSet:
+		for _, memberSet := range set.MemberIPSets {
+			contents = append(contents, memberSet.HashedName)
+		}
+		return contents, nil
+	default:
+		return contents, fmt.Errorf("Unknown set type %s", setType)
+	}
+}
+
+func getSetKind(set *IPSet) SetKind {
+	switch set.Type {
+	case CIDRBlocks:
+		return HashSet
+	case NameSpace:
+		return HashSet
+	case NamedPorts:
+		return HashSet
+	case KeyLabelOfPod:
+		return HashSet
+	case KeyValueLabelOfPod:
+		return HashSet
+	case KeyLabelOfNameSpace:
+		return ListSet
+	case KeyValueLabelOfNameSpace:
+		return ListSet
+	case NestedLabelOfPod:
+		return ListSet
+	default:
+		return "unknown"
+	}
+}
+
+func (set *IPSet) AddMemberIPSet(memberIPSet *IPSet) {
+	set.MemberIPSets[memberIPSet.Name] = memberIPSet
+}
+
+func (set *IPSet) IncIpsetReferCount() {
+	set.IpsetReferCount++
+}
+
+func (set *IPSet) DecIpsetReferCount() {
+	set.IpsetReferCount--
+}
+
+func (set *IPSet) AddSelectorReference(netPolName string) {
```

Comment:
different arg name (below func too)?

Reply:
essentially what we are adding is the network policy name if its a selector ref or a normal ref.